Test/HUnit/Lang.hs

```diff
@@ -89,4 +91,8 @@ performTestCase action =
        -- is not the case, please email the maintainer.
        E.Handler (\e -> throw (e :: E.AsyncException)),
 
+#if MIN_VERSION_base(4,9,0)
+       E.Handler (\(E.ErrorCallWithLocation msg _) -> return $ Error Nothing msg),
```

Comment:
@hvr I haven't had the bandwidth to follow the changes of GHC head.  We now have locations added to `ErrorCall`?  I'm puzzled what the reason is to add it there instead of to `SomeException`, so that all exceptions can have a source location (and / or call stack)?

To the change above, I haven't tried this, but would this effectively discard the source location where the error happened?  I'm puzzled whether this is useful from a users perspective.  My reasoning is that the source location can be useful to resolve the failing test.  But I haven't used this for TDD, so I'm not sure what I would prefer for my workflow.